cpp/src_prims/linalg/cublas_wrappers.h

```diff
@@ -181,6 +181,143 @@ inline cublasStatus_t cublasgemm(cublasHandle_t handle,
+}
+/** @} */
+
+/**
+ * @defgroup solverbatched cublas getrf/gettribatched calls
+ * @{
+ */
+
+template <typename T>
+cublasStatus_t cublasgetrfBatched(cublasHandle_t handle, int n,
+                                  T *const A[],    /*Device pointer*/
+                                  int lda, int *P, /*Device Pointer*/
+                                  int *info,       /*Device Pointer*/
+                                  int batchSize, cudaStream_t stream);
+
+template <>
+inline cublasStatus_t cublasgetrfBatched(cublasHandle_t handle, int n,
+                                         float *const A[], /*Device pointer*/
+                                         int lda, int *P,  /*Device Pointer*/
+                                         int *info,        /*Device Pointer*/
+                                         int batchSize, cudaStream_t stream) {
+  CUBLAS_CHECK(cublasSetStream(handle, stream));
+  return cublasSgetrfBatched(handle, n, A, lda, P, info, batchSize);
+}
+
+template <>
+inline cublasStatus_t cublasgetrfBatched(cublasHandle_t handle, int n,
+                                         double *const A[], /*Device pointer*/
+                                         int lda, int *P,   /*Device Pointer*/
+                                         int *info,         /*Device Pointer*/
+                                         int batchSize, cudaStream_t stream) {
+  CUBLAS_CHECK(cublasSetStream(handle, stream));
+  return cublasDgetrfBatched(handle, n, A, lda, P, info, batchSize);
+}
+
+template <typename T>
+cublasStatus_t cublasgetriBatched(cublasHandle_t handle, int n,
+                                  const T *const A[],    /*Device pointer*/
+                                  int lda, const int *P, /*Device pointer*/
+                                  T *const C[],          /*Device pointer*/
+                                  int ldc, int *info, int batchSize,
+                                  cudaStream_t stream);
+
+template <>
+inline cublasStatus_t cublasgetriBatched(
+  cublasHandle_t handle, int n, const float *const A[], /*Device pointer*/
+  int lda, const int *P,                                /*Device pointer*/
+  float *const C[],                                     /*Device pointer*/
+  int ldc, int *info, int batchSize, cudaStream_t stream) {
+  CUBLAS_CHECK(cublasSetStream(handle, stream));
+  return cublasSgetriBatched(handle, n, A, lda, P, C, ldc, info, batchSize);
+}
+
+template <>
+inline cublasStatus_t cublasgetriBatched(
+  cublasHandle_t handle, int n, const double *const A[], /*Device pointer*/
+  int lda, const int *P,                                 /*Device pointer*/
+  double *const C[],                                     /*Device pointer*/
+  int ldc, int *info, int batchSize, cudaStream_t stream) {
+  CUBLAS_CHECK(cublasSetStream(handle, stream));
+  return cublasDgetriBatched(handle, n, A, lda, P, C, ldc, info, batchSize);
```

Comment:
You're passing a `const double *const *` to a `const double **`. It should be an error (and my linter reports it as such), I'm surprised that it compiles.

Reply:
Update on this: it is actually fine because it matches cuBLAS headers and cuBLAS documentation is outdated. The issue is on cuBLAS side then.
You can resolve this discussion.